core/node/external_proof_integration_api/src/metrics.rs

```diff
@@ -0,0 +1,60 @@
+use std::time::Duration;
+
+use tokio::time::Instant;
+use vise::{Counter, EncodeLabelSet, EncodeLabelValue, Family, Histogram, LabeledFamily, Metrics};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet, EncodeLabelValue)]
+#[metrics(label = "outcome", rename_all = "snake_case")]
+pub(crate) enum CallOutcome {
+    Success,
+    Failure,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelSet, EncodeLabelValue)]
+#[metrics(label = "type", rename_all = "snake_case")]
+pub(crate) enum MethodType {
```

Comment:
nit: it's not really `MethodType`, it's just a `Method`, or `Endpoint`, or `MethodName`.